File: BookWorm/Models/Shipping.cs
namespace BookWorm.Models
{
    public class Shipping
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Address1 { get; set; }

        public string Address2 { get; set; }

        public string City { get; set; }

        public int ShoppingCartId { get; set; }

        public ShoppingCart ShoppingCart { get; set; }

    }
}

File: BookWorm/ClientApp/src/app/shared/models/shipping.ts
import { FormGroup } from '@angular/forms';
import { ShoppingCart } from './shopping-cart';

export class Shipping {
    id: number;
    name: string;
    address1: string;
    address2: string;
    city: string;
    shoppingCartId: number;
    appUserId: string;

    constructor(form: FormGroup, cart: ShoppingCart, user) {
        this.name = form.get('name').value;
        this.address1 = form.get('address1').value;
        this.address2 = form.get('address2').value;
        this.city = form.get('city').value;
        this.shoppingCartId = cart.id;
        this.appUserId = user.UserID;
    }
}

